executor/join.go

```diff
@@ -313,8 +317,77 @@ func (e *HashJoinExec) handleJoinWorkerPanic(r interface{}) {
 	e.joinWorkerWaitGroup.Done()
 }
 
+// handling unmatched rows from the hash table
+func (e *HashJoinExec) handleUnmatchedRowsFromHashTable(workerID uint, inMemory bool) {
+	ok, joinResult := e.getNewJoinResult(workerID)
+	if !ok {
+		return
+	}
+	if inMemory { // Concurrently Handling unmatched rows from the hash table that is in memory
+		numChks := e.rowContainer.records.NumChunks()
+		for i := int(workerID); i < numChks; i += int(e.concurrency) {
+			chk := e.rowContainer.records.GetChunk(i)
+			for j := 0; j < chk.NumRows(); j++ {
+				if e.outerMatchedStatus[i].UnsafeIsSet(j) == false { // process unmatched outer rows
+					e.joiners[workerID].onMissMatch(false, chk.GetRow(j), joinResult.chk)
+				}
+				if joinResult.chk.IsFull() {
+					e.joinResultCh <- joinResult
+					ok, joinResult = e.getNewJoinResult(workerID)
+					if !ok {
+						return
+					}
+				}
+			}
+		}
+	} else { // Sequentially handling unmatched rows from the hash table that is from disk
+		numChks := e.rowContainer.recordsInDisk.NumChunks()
+		for i := 0; i < numChks; i++ {
+			numOfRows := e.rowContainer.recordsInDisk.NumRowsOfChunk(i)
```

Comment:
It a little ugly here...
I think we can do the following things in an individual PR:
1. rename chunk.List to chunk.ListInMemory
2. define an interface 
``` go
type List interface{
    Add(chk *Chunk)
    NumChunks()
    GetRow(chunk.RowPtr)
}
```
3. remove the useless duplicate code for this function

@fzhedu @SunRunAway

Reply:
This change effects too much code, so regect this suggestion.